File: scripts/fix_use_client.py
import os

admin_dir = r"C:\Users\akaye\Desktop\PROjects\grasag-upsa\frontend\src\app\admin"
components_dir = r"C:\Users\akaye\Desktop\PROjects\grasag-upsa\frontend\src\components"

hooks = ["useState", "useEffect", "useForm", "useRouter", "useSearchParams"]

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    if any(hook in content for hook in hooks):
        if not content.strip().startswith("'use client'") and not content.strip().startswith('"use client"'):
            print(f"Adding 'use client' to {filepath}")
            # Insert at the very beginning
            new_content = "'use client';\n\n" + content
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(new_content)

for root, dirs, files in os.walk(admin_dir):
    for file in files:
        if file.endswith('.tsx'):
            process_file(os.path.join(root, file))

for root, dirs, files in os.walk(components_dir):
    for file in files:
        if file.endswith('.tsx'):
            process_file(os.path.join(root, file))

File: unused_archive/scripts/fix_use_client.js
const fs = require('fs');
const path = require('path');

const adminDir = path.resolve("C:/Users/akaye/Desktop/PROjects/grasag-upsa/frontend/src/app/admin");
const componentsDir = path.resolve("C:/Users/akaye/Desktop/PROjects/grasag-upsa/frontend/src/components");

const hooks = ["useState", "useEffect", "useForm", "useRouter", "useSearchParams"];

function processFile(filepath) {
  const content = fs.readFileSync(filepath, 'utf8');
  
  const hasHook = hooks.some(hook => content.includes(hook));
  const hasUseClient = content.trim().startsWith("'use client'") || content.trim().startsWith('"use client"');
  
  if (hasHook && !hasUseClient) {
    console.log(`Adding 'use client' to ${filepath}`);
    const newContent = "'use client';\n\n" + content;
    fs.writeFileSync(filepath, newContent, 'utf8');
  }
}

function walk(dir) {
  const list = fs.readdirSync(dir);
  list.forEach(file => {
    const fullpath = path.join(dir, file);
    const stat = fs.statSync(fullpath);
    if (stat.isDirectory()) {
      walk(fullpath);
    } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
      processFile(fullpath);
    }
  });
}

walk(adminDir);
walk(componentsDir);
console.log("Done checking files!");
